system/ui/lib/list_view.py

```diff
@@ -11,383 +9,289 @@
 from openpilot.system.ui.lib.toggle import Toggle, WIDTH as TOGGLE_WIDTH, HEIGHT as TOGGLE_HEIGHT
 from openpilot.system.ui.lib.widget import Widget
 
-ITEM_BASE_HEIGHT = 170
 LINE_PADDING = 40
-LINE_COLOR = rl.GRAY
+ITEM_BASE_HEIGHT = 170
 ITEM_PADDING = 20
-ITEM_SPACING = 80
 ITEM_TEXT_FONT_SIZE = 50
 ITEM_TEXT_COLOR = rl.WHITE
 ITEM_DESC_TEXT_COLOR = rl.Color(128, 128, 128, 255)
 ITEM_DESC_FONT_SIZE = 40
 ITEM_DESC_V_OFFSET = 140
-RIGHT_ITEM_PADDING = 20
 ICON_SIZE = 80
 BUTTON_WIDTH = 250
 BUTTON_HEIGHT = 100
-BUTTON_BORDER_RADIUS = 50
 BUTTON_FONT_SIZE = 35
-BUTTON_FONT_WEIGHT = FontWeight.MEDIUM
 
-TEXT_PADDING = 20
+
+# Type Aliases for Clarity
+_StrSrc = str | Callable[[], str] | None
+_BoolSrc = bool | Callable[[], bool]
 
 
-def _resolve_value(value, default=""):
+def _get_value(value, default=""):
   if callable(value):
     return value()
   return value if value is not None else default
 
 
-# Abstract base class for right-side items
-class ItemAction(Widget, ABC):
-  def __init__(self, width: int = 100, enabled: bool | Callable[[], bool] = True):
+class BaseListItem(Widget):
+  """Base class for list items"""
+
+  def __init__(self):
+    super().__init__()
+
+  def is_visible(self):
+    return True
+
+  def _get_height(self, max_width: int) -> float:
+    raise NotImplementedError("Subclasses must implement the get_height method")
```

Comment:
I was thinking something along the lines of the Widget class implementing this as returning self.rect.height by default or similar or a NotImplementedError

Reply:
y, put to widget class is better, we can remove this base class.